contracts/libraries/CurrencyLibrary.sol

```diff
@@ -32,25 +32,27 @@ library CurrencyLibrary {
             if (!success) revert NativeTransferFailed();
         } else {
             assembly {
-                // Get a pointer to some free memory.
-                let freeMemoryPointer := mload(0x40)
+                // We'll write our calldata to this slot below, but restore it later.
+                let memPointer := mload(0x40)
 
                 // Write the abi-encoded calldata into memory, beginning with the function selector.
-                mstore(freeMemoryPointer, 0xa9059cbb00000000000000000000000000000000000000000000000000000000)
-                mstore(add(freeMemoryPointer, 4), to) // Append the "to" argument.
-                mstore(add(freeMemoryPointer, 36), amount) // Append the "amount" argument.
+                mstore(0, 0xa9059cbb00000000000000000000000000000000000000000000000000000000)
+                mstore(4, to) // Append the "to" argument.
+                mstore(36, amount) // Append the "amount" argument.
 
                 success :=
                     and(
                         // Set success to whether the call reverted, if not we check it either
                         // returned exactly 1 (can't just be non-zero data), or had no return data.
                         or(and(eq(mload(0), 1), gt(returndatasize(), 31)), iszero(returndatasize())),
-                        // We use 68 because the length of our calldata totals up like so: 4 + 32 * 2.
-                        // We use 0 and 32 to copy up to 32 bytes of return data into the scratch space.
-                        // Counterintuitively, this call must be positioned second to the or() call in the
-                        // surrounding and() call or else returndatasize() will be zero during the computation.
-                        call(gas(), currency, 0, freeMemoryPointer, 68, 0, 32)
+                        // We use 68 because that's the total length of our calldata (4 + 32 * 2)
+                        // Counterintuitively, this call() must be positioned after the or() in the
+                        // surrounding and() because and() evaluates its arguments from right to left.
+                        call(gas(), currency, 0, 0, 68, 0, 32)
                     )
+
+                mstore(0x60, 0) // Restore the zero slot to zero.
```

Comment:
I don't think memory slot `0x60` is tampered with so it isn't necessary to clean. Besides IMHO Solmate is lagging behind Solady in terms of gas efficiency. Solady's [implementation](https://github.com/Vectorized/solady/blob/37b4fd04dd82977eb0161839389187abb7afdb58/src/utils/SafeTransferLib.sol#L247-L252) removes the unnecessary `mstore(0x60, 0)` and `gt(returndatasize(), 31)`, which hasn't been merged to Solmate since a year ago. If I were to insert the code instead of importing a library to use directly, I would opt for the better one.

Reply:
thanks for your comment @shuhuiluo! For now we're keeping in these lines even if they arent needed, as this code is an exact copy of a version of solmate. However, if you think that we should be changing this library to use solady instead, please feel free to put in a new PR that does so :)